src/main/java/ch/njol/skript/effects/EffPlaySound.java

```diff
@@ -47,186 +46,142 @@
 		"are supported. Playing resource pack sounds are supported too. The sound category is 'master' by default. ",
 		"",
 		"Please note that sound names can get changed in any Minecraft or Spigot version, or even removed from Minecraft itself."})
-@Examples({"play sound \"block.note_block.pling\" # It is block.note.pling in 1.12.2",
-		"play sound \"entity.experience_orb.pickup\" with volume 0.5 to the player",
-		"play sound \"custom.music.1\" in jukebox category at {speakerBlock}"})
+@Examples({
+	"play sound \"block.note_block.pling\" # It is block.note.pling in 1.12.2",
+	"play sound \"entity.experience_orb.pickup\" with volume 0.5 to the player",
+	"play sound \"custom.music.1\" in jukebox category at {speakerBlock}"
+})
 @Since("2.2-dev28, 2.4 (sound categories)")
-@RequiredPlugins("Minecraft 1.11+ (sound categories)")
 public class EffPlaySound extends Effect {
 
-	private static final boolean SOUND_CATEGORIES_EXIST = Skript.classExists("org.bukkit.SoundCategory");
-	private static final Pattern SOUND_VALID_PATTERN = Pattern.compile("[a-z0-9\\/:._-]+"); // Minecraft only accepts these characters
+	private static final Pattern KEY_PATTERN = Pattern.compile("([a-z0-9._-]+:)?[a-z0-9/._-]+");
 	
 	static {
-		if (SOUND_CATEGORIES_EXIST) {
-			Skript.registerEffect(EffPlaySound.class,
-					"play sound[s] %strings% [(in|from) %-soundcategory%] " +
-							"[(at|with) volume %-number%] [(and|at|with) pitch %-number%] at %locations% [(to|for) %-players%]",
-					"play sound[s] %strings% [(in|from) %-soundcategory%] " +
-							"[(at|with) volume %-number%] [(and|at|with) pitch %-number%] [(to|for) %players%] [(at|from) %-locations%]");
-		} else {
-			Skript.registerEffect(EffPlaySound.class,
-					"play sound[s] %strings% [(at|with) volume %-number%] " +
-							"[(and|at|with) pitch %-number%] at %locations% [(to|for) %-players%]",
-					"play sound[s] %strings% [(at|with) volume %-number%] " +
-							"[(and|at|with) pitch %-number%] [(to|for) %players%] [(at|from) %-locations%]");
-		}
+		Skript.registerEffect(EffPlaySound.class,
```

Comment:
You should add support for this https://helpch.at/docs/1.18.2/org/bukkit/entity/Player.html#playSound(org.bukkit.entity.Entity,org.bukkit.Sound,float,float) method while you're at it (it's 1.18+ btw) and I'm thinking the syntax could look a bit like this `play sound %string% **on** %entity% ... `, it basically plays a sound that dynamically follows the specified entity; playing the sound at its location.

Reply:
[Entity Sound Effect](https://wiki.vg/Protocol#Entity_Sound_Effect) is severely limited since it only supports vanilla sound effects to be played, and the player interface only support its own sound enum for it (so no sounds from namespaced resources). Once those issues get resolved (probably by integrating adventure), I will be happy to pursue this in another PR.